primitives/transaction/src/lib.rs

```diff
@@ -135,16 +138,199 @@ impl SignatureProof {
     }
 }
 
-impl Default for SignatureProof {
+impl Default for EdDSASignatureProof {
     fn default() -> Self {
-        SignatureProof {
+        EdDSASignatureProof {
             public_key: Default::default(),
             merkle_path: Default::default(),
             signature: Signature::from_bytes(&[0u8; Signature::SIZE]).unwrap(),
         }
     }
 }
 
+bitflags! {
+    #[derive(Default, Serialize, Deserialize)]
+    /// Some authenticators may behave non-standard for signing with Webauthn:
+    ///
+    /// - they might not include the mandatory `crossOrigin` field in clientDataJSON
+    /// - they might escape the `origin`'s forward slashes with backslashes, although not necessary for UTF-8 nor JSON encoding
+    ///
+    /// To allow the WebauthnSignatureProof to construct a correct `clientDataJSON` for verification,
+    /// the proof needs to know these non-standard behaviors.
+    ///
+    /// See this tracking issue for Android Chrome: https://bugs.chromium.org/p/chromium/issues/detail?id=1233616
+    pub struct WebauthnClientDataFlags: u8 {
+        const NO_CROSSORIGIN_FIELD  = 1 << 0;
+        const ESCAPED_ORIGIN_SLASHES = 1 << 1;
+
+        // const HAS_EXTRA_FIELDS = 1 << 7; // TODO Replace client_data_extra_fields length null byte when no extra fields are present
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct WebauthnSignatureProof {
+    pub public_key: WebauthnPublicKey,
+    pub merkle_path: Blake2bMerklePath,
+    pub signature: Signature,
+    pub host: String,
```

Comment:
In practical terms, what would the host be? Is it `nimiq.com` for the wallet? Isn't that a privacy concern as one can now identify which wallet has been used?

Reply:
Actually, you can probably also infer things about the browser etc.

If we think that this is an issue, I have some thoughts about how to work around it.

Reply:
You already leak information about your wallet and browser when connecting to the network through user-agent and origin headers in websocket connections.

Reply:
Yes. The data over which webauthn makes the signature includes the origin. Host is the origin without protocol, so `wallet.nimiq.com` in our case.

If we would omit any identifying information from the proof, we'd have to include the authenticatorData and clientDataJSON directly, which 1. takes more bytes on average and does not let us verify that the challenge is actually the tx hash.

So if you have ideas how to avoid, please share.

@mar-v-in I'd say there's a difference between the two, as websocket info is only shared with nodes and not put on-chain to be visible by anybody.

Reply:
As mentioned on our call, we could avoid putting the data on-chain using a zk-proof (in the future).